gammapy/cube/simulate.py

```diff
@@ -321,27 +321,43 @@ def event_list_meta(dataset, observation):
     def run(self, dataset, observation=None):
         """Run the event sampler, applying IRF corrections.
 
-        Parameters
-        ----------
-        dataset : `~gammapy.cube.MapDataset`
-            Map dataset
-        observation : `~gammapy.data.Observation`
-            In memory observation.
-        edisp : Bool
-            It allows to include or exclude the Edisp in the simulation.
-
-        Returns
-        -------
-        events : `~gammapy.data.EventList`
-            Event list.
+            Parameters
+            ----------
+            dataset : `~gammapy.cube.MapDataset`
+                Map dataset
+            observation : `~gammapy.data.Observation`
+                In memory observation.
+            edisp : Bool
+                It allows to include or exclude the Edisp in the simulation.
+
+            Returns
+            -------
+            events : `~gammapy.data.EventList`
+                Event list.
         """
-        events_bkg = self.sample_background(dataset)
         events_src = self.sample_sources(dataset)
-        events_src = self.sample_psf(dataset.psf, events_src)
 
-        events_src = self.sample_edisp(dataset.edisp, events_src)
+        if dataset.psf:
+            events_src = self.sample_psf(dataset.psf, events_src)
+        else:
+            events_src.table.add_columns(
```

Comment:
Maybe just use:
```
event_src.table["RA"] = event_src.table["RA_TRUE"]
event_src.table["DEC"] = event_src.table["DEC_TRUE"]
```
Seems easier to read...

Reply:
OK